### DISCUSSION TITLE
odd MX record creating wording and results.

### DISCUSSION THREAD

**Discussion by kollross** [sys4/netbox-plugin-dns#263] [Q&A] [answered]
**Versions**
NetBox Version: 4.0.1
NetBox DNS Version: 1.0
Python Version: 3.10.12

**Describe the bug**
When creating MX records (unmanaged), unclear about what information goes into the Name field (which is required) .   Putting something like "mx.mydomain.com", produces something like this in the BIND file "mx.mydomain.com.mydomain.com	86400	IN	MX	100 mx.mydomain.com.

**To Reproduce**
1. Go to 'Zones'
2. Click on 'Add new record'
3. Something in Name field. 

**Expected result**
Unclear of what format it should be in.

**Actual result**

**Comment by peteeckel** [comment, marked as answer]
Hi @kollross,

* The name of the MX record is usually the name of the domain itself, i.e. the empty label `@`. YMMV, but that should be correct in the majority of the cases.
* A label without a trailing dot `.` is interpreted as relative to the domain name, i.e. `mx.mydomain.com`would be interpreted as `mx.mydomain.com.mydomain.com` if you add it to the zone `mydomain.com`. Usually not what you want :-)
* Even if, the `mx.mydomain.com` would almost certainly be wrong, as you usually don't want to define an MX for the domain `mx.mydomain.com` but for `mydomain.com`.
* All of this is standard DNS functionality, so it's not an issue :-)

So, the correct entry would be:
```
@    IN    MX    10    mx1.mydomain.com.
```
or
```
mydomain.com.    IN    MX    10    mx1.mydomain.com.
```

**Reply by kollross** [reply]
The @ was the missing piece of the puzzle.   The title "Name" is somewhat vague on what information is actually needed for that line.

**Reply by peteeckel** [reply]
Maybe you're right, but that's unfortunately the way it is named throughout all of the relevant RFCs, so giving it a different name would have been even more confusing ...
